lib/charms/mysql/v0/mysql.py

```diff
@@ -772,28 +775,40 @@ def render_myqld_configuration(
             max_connections = self.get_max_connections(available_memory)
             if available_memory < 2 * BYTES_1GiB:
                 # disable memory instruments if we have less than 2GiB of RAM
-                disable_memory_instruments = "performance-schema-instrument = 'memory/%=OFF'"
-
-        content = [
-            "[mysqld]",
-            "bind-address = 0.0.0.0",
-            "mysqlx-bind-address = 0.0.0.0",
-            f"report_host = {self.instance_address}",
-            f"max_connections = {max_connections}",
-            f"innodb_buffer_pool_size = {innodb_buffer_pool_size}",
-        ]
-        if innodb_buffer_pool_chunk_size:
-            content.append(f"innodb_buffer_pool_chunk_size = {innodb_buffer_pool_chunk_size}")
+                performance_schema_instrument = "'memory/%=OFF'"
+
+        config = configparser.ConfigParser(interpolation=None)
+
+        config["mysqld"] = {
+            "bind-address": "0.0.0.0",
+            "mysqlx-bind-address": "0.0.0.0",
+            "report_host": self.instance_address,
+            "max_connections": f"{ max_connections }",
+            "innodb_buffer_pool_size": f"{ innodb_buffer_pool_size }",
+            "log_error_services": "log_filter_internal;log_sink_internal",
+            "log_error": f"{ CHARMED_MYSQL_COMMON_DIRECTORY }/var/log/mysql/error.log",
+            "general_log": "ON",
+            "general_log_file": f"{ CHARMED_MYSQL_COMMON_DIRECTORY }/var/log/mysql/general.log",
+            "slow_query_log": "ON",
+            "slow_query_log_file": f"{CHARMED_MYSQL_COMMON_DIRECTORY }/var/log/mysql/slowquery.log",
+        }
 
-        if disable_memory_instruments:
-            content.append(disable_memory_instruments)
+        if innodb_buffer_pool_chunk_size:
+            config["mysqld"][
+                "innodb_buffer_pool_chunk_size"
+            ] = f"{ innodb_buffer_pool_chunk_size }"
+        if performance_schema_instrument:
+            config["mysqld"]["performance-schema-instrument"] = performance_schema_instrument
         if group_replication_message_cache_size:
-            content.append(
-                f"loose-group_replication_message_cache_size = {group_replication_message_cache_size}"
-            )
+            config["mysqld"][
+                "loose-group_replication_message_cache_size"
+            ] = f"{ group_replication_message_cache_size }"
 
-        content.append("\n")
-        return "\n".join(content)
+        string_io = io.StringIO()
+        config.write(string_io)
+        output = string_io.getvalue()
+        string_io.close()
+        return output
```

Comment:
this is indeed written to a file later (in the calling method). the issue is, since this is a charm lib, i'd have to move furniture around in the k8s charm as well which i thought i would avoid doing so unnecessarily. therefore, i am returning a string representing the mysqld config (as it was before this refactor)

Reply:
using `with` in 2d10ae227f44160411aacfdd7f96eea2ef5fdec5